sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateMutableProjection.scala

```diff
@@ -45,7 +45,37 @@ object GenerateMutableProjection extends CodeGenerator[Seq[Expression], () => Mu
           else
             ${ctx.setColumn("mutableRow", e.dataType, i, evaluationCode.primitive)};
         """
-    }.mkString("\n")
+    }
+    val projectionBlocks = projectionCode.foldLeft(List.empty[String]) {
```

Comment:
would be great if you can rewrite the code to make it slightly less functional here. I'd actually prefer a more imperative version here